definitions/divorce/json/CaseEvent/CaseEvent-nonprod.json

```diff
@@ -9,5 +9,31 @@
     "PreConditionState(s)": "AwaitingServicePayment",
     "PostConditionState": "AwaitingServiceConsideration",
     "SecurityClassification": "Public"
+  },
+  {
+    "LiveFrom": "01/01/2019",
+    "CaseTypeID": "DIVORCE",
+    "ID": "allowPetitionerToAmend",
+    "Name": "Allow Petitioner to amend",
+    "Description": "Allow Petitioner to amend",
+    "DisplayOrder": 23,
+    "PreConditionState(s)": "ServiceApplicationNotApproved;Issued;Rejected;PendingRejection;AosAwaiting;AosStarted;AosSubmittedAwaitingAnswer;AwaitingDecreeNisi;AosOverdue;AwaitingReissue;AosCompleted;AosAwaitingSol;AosPreSubmit;AosDrafted;AwaitingService",
+    "PostConditionState": "AwaitingAmendCase",
+    "CallBackURLSubmittedEvent": "${CCD_DEF_COS_URL}/amend-application",
+    "RetriesTimeoutURLSubmittedEvent": "120,120",
+    "SecurityClassification": "Public"
+  },
+  {
+    "LiveFrom": "29/06/2020",
+    "CaseTypeID": "DIVORCE",
+    "ID": "amendPetitionForRefusalSol",
+    "Name": "Submit amended application",
+    "Description": "Submit amended application",
+    "DisplayOrder": 3,
+    "PreConditionState(s)": "ServiceApplicationNotApproved;DNisRefused;Issued;Rejected;PendingRejection;AosAwaiting;AosStarted;AosSubmittedAwaitingAnswer;AwaitingDecreeNisi;AosOverdue;AwaitingReissue;AosCompleted;AosAwaitingSol;AosPreSubmit;AosDrafted;AwaitingService;AwaitingAmendCase",
+    "PostConditionState": "AmendPetition",
+    "CallBackURLAboutToSubmitEvent": "${CCD_DEF_COS_URL}/solicitor-amend-petition-dn-rejection",
+    "RetriesTimeoutURLAboutToSubmitEvent": "120,120",
+    "SecurityClassification": "Public"
   }
 ]
```

Comment:
As per AC's for this ticket `allowPetitionerToAmend` and `amendPetitionForRefusalSol` pre-state are to be extended to include `ServiceApplicationNotApproved`.  Issue i faced was that these events are part of the `pet-amend` epic and the new pre-state is part of `deemed-and-dispensed` epic. I had originally separated them in respective feature files but CCD build was failing as it saw duplicate events in the definition. My solution was to move them to the `nonprod` file. Seeing that the service application is part of another event, if the `pet-amentd` event gets released first the the `ServiceApplicationNotApproved` in this definition would go also. This might not be an issue as the `RBAC` for this state are in the right feature files i.e. the `deemed-and-dispensed` one, just though i'd mention this here explain why this definition is in this file and not either `pet-amend` or `deemed-and-dispensed` feature files and to draw your attention to it. @yarpo @Dizrupto please advise.

Reply:
You need to wait until I merge this: https://github.com/hmcts/div-ccd-definitions/pull/483

Nothing else we can do.

Reply:
Yep that's a good point. I guess when Patryk's PR is merged you can move it to the 'deemed-and-dispensed' non-prod file so we make sure to release it at the same time.

Reply:
Ok, thanks @Dizrupto